_posts/2021-01-18-changelog-139.markdown

```diff
@@ -0,0 +1,115 @@
+      Such test attributes are quite complex to handle in the IDE because we do not want to gray out all the tests, but do want to gray out only the mock objects.
+      So the current logic behind that is to gray out "test" code only when "mocking", i.e. when there is a `cfg(not(test))` somewhere in the code.
+      Note, this new logic currently affects the external dependencies only, excluding your workspace, in order not to break down the code insight for your tests.
+      TODO: might be in the features block
+
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6574 -->
+    * Fix resolution of Cargo features for aliased packages like `foo_alias = { package = "foo", path = "../foo"}`
+
+* [New name resolution](https://github.com/intellij-rust/intellij-rust/issues/6217) engine improvements:
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6576 -->
+    * Support `#![macro_use]` attribute at file level
+
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6543 -->
+    * Consider [textual scoping](https://doc.rust-lang.org/reference/macros-by-example.html#textual-scope) when resolving macros
+
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6568 -->
+    * Properly process items with restricted visibility that contains an absolute path, e.g. `pub(in crate::inner1)`
+
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6566 -->
+    * Fix name resolution of qualified macro calls inside items in some cases
+
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6545 -->
+    * Properly handle renaming of enum variants
+
+  Note, you can enable the new engine via `Use experimental name resolution engine` option in `Preferences | Languages & Frameworks | Rust` settings
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6606 -->
+* Fix name resolution and completion of `Self` keyword for nested structs, enums, and `impl` blocks
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6613 -->
+* Improve the new `Fill function arguments` quick-fix (by [@Kobzol])
+    * Support filling arguments of a closure call
+    * Take actual types into account when constructing default values of generic types
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6515 -->
+* `Remove parameter` quick-fix no longer removes references to the modified function (by [@Kobzol])
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6588 -->
+* Fix `Trait implementation` inspection on `impl` blocks containing macro-expanded members
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6541 -->
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6542 -->
+* Improve parser recovery and syntax error messages for incomplete paths
+
+## Internal Improvements
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6578 -->
+* Introduce `TomlVisitor` (by [@Kobzol])
```

Comment:
Let's add a short explanation why it's important for developers

Reply:
Added `This class would be used to implement inspections that process PSI elements of a `Cargo.toml` file`